extension/builtin/default_registry.go

```diff
@@ -36,62 +29,20 @@ var (
 	MySQLURIFlag = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test", "uri to use with mysql storage")
 )
 
-// Default implementation of extension.Registry.
-type defaultRegistry struct {
-	db *sql.DB
+// UseSQLDB configures the registry to use the given SQL database for all storage.
+func UseSQLDB(r *extension.Registry, db *sql.DB) {
+	r.AdminStorage = mysql.NewAdminStorage(db)
+	r.LogStorage = mysql.NewLogStorage(db)
+	r.MapStorage = mysql.NewMapStorage(db)
 }
 
-func (r *defaultRegistry) GetAdminStorage() storage.AdminStorage {
-	return mysql.NewAdminStorage(r.db)
-}
-
-func (r *defaultRegistry) GetLogStorage() (storage.LogStorage, error) {
-	return mysql.NewLogStorage(r.db), nil
-}
-
-func (r *defaultRegistry) GetMapStorage() (storage.MapStorage, error) {
-	return mysql.NewMapStorage(r.db), nil
-}
-
-func (r *defaultRegistry) GetKeyProvider() (keys.Provider, error) {
-	return keyProvider{}, nil
-}
-
-// NewExtensionRegistry returns an extension.Registry implementation backed by a given
-// MySQL database. It supports loading private keys from PEM files.
-func NewExtensionRegistry(db *sql.DB) (extension.Registry, error) {
-	return &defaultRegistry{db: db}, nil
-}
-
-// NewDefaultExtensionRegistry returns the default extension.Registry implementation, which is
-// backed by a MySQL database and configured via flags.
-// It supports loading private keys from PEM files.
-func NewDefaultExtensionRegistry() (extension.Registry, error) {
+// UseMySQLDBSetByFlags configures the registry to use the MySQL database specified by the --mysql_uri flag for all storage.
+func UseMySQLDBSetByFlags(r *extension.Registry) error {
 	db, err := mysql.OpenDB(*MySQLURIFlag)
```

Comment:
Food for thought: because we create the DB object inside this func, and the registry then takes indirect ownership of it, we never call db.Close(). Ideally, that should be a defer call inside main(). We didn't do it properly before, but we still don't. I'm not sure how important it is to call db.Close() either (probably not that much, unless we're leaking resources and having processes die/restart all the time).

Maybe that's an argument for killing this entire file and moving the logic to `main.go` files everywhere? Thoughts?